### DISCUSSION TITLE
HardhatError: HH1007: The file  is treated as local but is outside the project

### DISCUSSION THREAD

**Discussion by hammersharkfish** [NomicFoundation/hardhat#4999] [General]
```
const hre = require("hardhat");

async function main() {
  // Run the compile task
  await hre.run("compile");

  console.log("Compilation finished successfully");
}

main()
  .then(() => process.exit(0))
  .catch((error) => {
    console.error(error);
    process.exit(1);
  });
```
`HardhatError: HH1007: The file some/outside/path/contracts/storage.sol is treated as local but is outside the 
project`
Here is my `path` of `hardhat.config.cjs` (of local directory)

```
  paths: {
    sources: 'some/path/outside/contracts',
    tests: ' some/path/outside/tests',
    cache: 'some/path/outside/cache',
    artifacts: 'some/path/outside/artifacts',
  }
```
My understanding is `paths` is used to find `sources` from any directory so why this error .

**Comment by schaable** [comment]
Hi @hammersharkfish, defining paths outside the root directory is not supported due to the complexities it introduces to the compilation process.
